### PR TITLE
feat: add more logs on dagger loading

### PR DESCRIPTION
**Pull request by TomChv**
Signed-off-by: Vasek - Tom C <tom@epitech.eu>

#Related to: #4820

### CODE REVIEW

**Comment by TomChv** [comment]
Another improvement could be to add logs directly in each SDK but:
- It's not a generic solution AKA we need to maintain each logs in each SDK
- We do not have access to progrock in SDK's so we might need to use the standard library to log, which will be great for user that use Dagger as an external library but these lines will not appears for these who use `dagger do` or `dagger run`

WDYT? @gerhard @vito @sipsma

File: engine/engine.go
Comment (TomChv) [reply engine/engine.go:142]: Thanks for your comment!
I think I'll refactor to use the first option since these tasks shall not take too much time!

Comment (TomChv) [reply engine/engine.go:142]: @vito With the second method, logs are not displayed :(
```
┣─┼─╮                                                                           
│ │ ▼ start engine                                                              
│ │ █ [0.21s] loading                                                           
│ │ ┻          
```

Comment (vito) [reply engine/engine.go]: @TomChv What you're seeing is probably because log output is hidden for completed vertices. If something is stuck or taking a while you'll see the logs, which seems to be the main goal of #4820. Sub-tasks will show up even after completion, but I think that might just be added noise, since most of the time you don't really care about the engine bootstrapping process.

**Comment by vito** [comment]
Also, I think you're probably already aware but this unfortunately doesn't cover what is probably the root of the issue, which is downloading the engine image. There's a goofy circular dependency there because we include the engine name in the Progrock root group labels, so we can't even show progress until the engine image is fetched. Would definitely like to refactor this a bit.

One option could be to simply initialize the root recorder without any labels, and then manually create another group with all the labels after the engine is created. I haven't thought too much about this, not sure if that breaks anything.

**Comment by gerhard** [comment]
Hey @TomChv, how is this coming along? We would love to get this into `v0.6.4` which is due this week.

**Comment by TomChv** [comment]
> Hey @TomChv, how is this coming along? We would love to get this into v0.6.4 which is due this week.

#5488 makes more sense for now according to https://github.com/dagger/dagger/pull/5436#issuecomment-1636348915

**Comment by gerhard** [comment]
Should this be closed now @TomChv?

**Comment by TomChv** [comment]
> https://github.com/dagger/dagger/pull/5582

No, I'll continue this PR, we're close to have a nice logging to users

**Comment by TomChv** [comment]
New display:

```shell
▼ start engine
█ [1.19s] loading
┃ Configure cache... OK!                                                                                                                                                              
┃ Create buildkit client...OK!                                                                                                                                                        
┃ Start session... OK!                                                                                                                                                                
┃ Connect to session... OK!   
```

**Comment by TomChv** [comment]
I do not understand why the Elixir PR is keep failing, it doesn't seem related to the PR: https://github.com/dagger/dagger/actions/runs/5811871350/job/15785950820?pr=5436#step:4:2269

**Comment by gerhard** [comment]
`test / sdk` is passing on `main`: https://github.com/dagger/dagger/actions/runs/5821926562/job/15785392662

Does it still fail if you rebase on `main` @TomChv?

**Comment by vito** [comment]
Currently tinkering with this; I really think it should be using tasks for the incremental progress steps (it just feels wrong not to, because that's exactly what they're for), and it could use a bit more wordsmithing (I don't think we want to expose "buildkit" for example).

I'll push my changes directly to this PR for expediency, and rebase while I'm at it to see if that fixes the Elixir issue

**Comment by vito** [comment]
Here's what I ended up with:

```
❯ dagger run ls
╮ 
▽ init
█ [0.66s] connect
┣ [0.44s] starting engine
┣ [0.14s] starting session
┻ 
█ [0.00s] ls
┃ auth
┃ bin
┃ buildkit
```

**Comment by TomChv** [comment]
@gerhard Some of your commit broke the DCO

> Commit sha: [c6adb97](https://github.com/dagger/dagger/pull/5436/commits/c6adb979032774111e2e91c664380b1a75684da0), Author: Gerhard Lazu, Committer: Vasek - Tom C; Expected "Gerhard Lazu [gerhard@users.noreply.github.com](mailto:gerhard@users.noreply.github.com)", but got "Gerhard Lazu [gerhard@dagger.io](mailto:gerhard@dagger.io)".
Commit sha: [b1fc56d](https://github.com/dagger/dagger/pull/5436/commits/b1fc56d4b84a6c13a591425c5f4799c672e15eb4), Author: Gerhard Lazu, Committer: Vasek - Tom C; Expected "Gerhard Lazu [gerhard@users.noreply.github.com](mailto:gerhard@users.noreply.github.com)", but got "Gerhard Lazu [gerhard@dagger.io](mailto:gerhard@dagger.io)".

How can I fix it? I think you need to rebase the PR yourself to correct the DCO so we can merge the PR

**Review by vito — APPROVED** [review]

**Comment by wingyplus** [comment]
> I do not understand why the Elixir PR is keep failing, it doesn't seem related to the PR: https://github.com/dagger/dagger/actions/runs/5811871350/job/15785950820?pr=5436#step:4:2269

I’m not in front of the notebook right now but my assumption is that the integration tests in the sdk that test against current engine (build against this PR) is failure somehow.

**Comment by gerhard** [comment]
Looking at this one now.

**Comment by TomChv** [comment]
@gerhard Everything is fine, we can merge this PR anytime now

**Review by gerhard — APPROVED** [review]
Looks great @vito!

We should settle on the same wording between SDKs & the CLI, but this is good for now. FWIW:

SDK output:
image

CLI output after this gets merged:
image

It's all good, we can do this in follow-up PRs 🚀

**Pull request merged**
